airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -266,6 +280,71 @@ def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         params = {"include": "description"}
         yield from self.read(partial(self._api_get, url="tickets"), params=params)
 
+    @staticmethod
+    def get_tickets(
+        result_return_limit: int, getter: Callable, params: Mapping[str, Any] = None, ticket_paginate_limit: int = 300
+    ) -> Iterator:
+        """
+        Read using getter
+
+        This block extends TicketsAPI Stream to overcome '300 page' server error.
+        Since the TicketsAPI Stream list has a 300 page pagination limit, after 300 pages, update the parameters with
+        query using 'updated_since' = last_record, if there is more data remaining.
+        """
+        params = params or {}
+
+        # Start page
+        page = 1
+        # Initial request parameters
+        params = {
+            **params, 
+            "order_type": "asc", # ASC order, to get the old records first
+            "order_by": "updated_at", 
+            "per_page": result_return_limit
+        }
+
+        while True:
+            params["page"] = page
+            batch = list(getter(params=params))
+            yield from batch
+
+            if len(batch) < result_return_limit:
+                return iter(())
+
+            # checkpoint & switch the pagination
+            if page == ticket_paginate_limit:
+                # get last_record from latest batch, pos. -1, because of ACS order of records
+                last_record_updated_at = batch[-1]["updated_at"]
+                page = 0  # reset page counter
+                # Adding +1 second to last_record value, to avoid record duplication,
+                # this potentialy could lead to skipping some records during the fetch,
+                # in cases where multiple records are created at the same datetime, at the level of seconds
+                # this behaviour is tested in test_300_page.py unit_test (test1)
+                # to avoid such ocations, please comment or remove the `.add(seconds=1)' part.
+                last_record_updated_at = pendulum.parse(last_record_updated_at).add(seconds=1)
```

Comment:
we shouldn't add 1 -- duplicating records is OK since they get deduplicated in the destination (airbyte is at-least-once delivery) but skipping records is data loss.

Reply:
Done.